console: Add common aliases for console methods

aks- opened the pull request:
- add common aliases as per following spec
- https://github.com/DeveloperToolsWG/console-object/blob/master/api.md

silverwind commented on doc/api/console.markdown:
```diff
@@ -48,6 +48,10 @@ is used on each argument.  See [util.format()][] for more information.
 
 Same as `console.log`.
 
+### console.debug([data][, ...])
```
I suggest using the same signature as in the spec: `console.debug(object [,object, ...])`

aks- replied:
Then we might want to change the complete doc for console. Because the signature to methods like log are also not matching with spec we are following https://github.com/nodejs/node/blob/master/doc/api/console.markdown#consolelogdata- Do you think I should change those them as well?

silverwind replied:
Yes, I think changing the signatures would be preferred. Can you do so in a second commit?

silverwind commented on doc/api/console.markdown:
```diff
@@ -56,6 +60,10 @@ Same as `console.log` but prints to stderr.
 
 Same as `console.error`.
 
+### console.exception([data][, ...])
```
Would also suggest using a similar `console.exception(object [, object, ...])` here.

silverwind commented on doc/api/console.markdown:
```diff
@@ -56,6 +60,10 @@ Same as `console.log` but prints to stderr.
 
+### console.exception([data][, ...])
+
+Same as `console.error`
```
Missing `.`

aks- replied:
Oops will fix it

silverwind commented:
Please also add some tests for these methods [here](https://github.com/nodejs/node/blob/master/test/parallel/test-console-instance.js).

jasnell commented:
Wouldn't this need to be `semver-major` as it could break anyone who has monkey patched these?

silverwind commented:
I don't think we have to support broken polyfills or monkey patches, and I think it's unlikely anyone would do so in the case of these methods.

aredridel commented:
Hm. on the converse side, this lets things designed for the browser run as is, even if they use these features.

chrisdickinson commented:
@jasnell:

> Wouldn't this need to be semver-major as it could break anyone who has monkey patched these? (same issue as nooping the undefined ones)

It seems low risk. If code is monkeypatching the `console` object, that code's new `debug`/`exception` methods should (continue to) win. I might be missing something though — do you have a link to the problems arising from `noop`'ing handy?

aks- commented:
@silverwind I have added the tests for this and changed the signture of newly added aliases as per browser spec. Besides that I created a new PR https://github.com/nodejs/node/pull/3584 in which I changed the method signatures which were there previously in doc to follow the browser spec. I haven't changed the signatures of methods which have differnent behaviour in node than in browser (e.g. console.assert). Now I am in confusion if we are actually following the spec?

silverwind commented:
@aks- thanks, I'll have a look later.

This PR is LGTM.

silverwind commented:
@aks- on second look, what's up with ` tools/v8-prof/tick-processor-tmp-1221` in that patch?

aks- commented:
@silverwind sorry I added that by accident I guess...removed it now

silverwind commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/665/

silverwind commented:
LGTM, will likely land this tomorrow.

Fishrock123 commented:
I'm not sure this is a good idea actually if we aren't going to follow the rest of the "spec".

@aks- Sorry, my fault for pulling you into this console-spec thing. :/

silverwind commented:
@Fishrock123 I think we should implement as much of this "spec" as is feasible. What's wrong with these alias methods?

Fishrock123 commented:
Nothing particularly but it's more things we need to keep supporting. Seems low-value -- stuff works fine if it follows this part but not another part kind of thing.

aks- commented:
@Fishrock123 it's okay, I am actually learning a lot about the structure of node codebase :)

silverwind commented:
Guess I'll put this on hold then.

Interestingly, [MDN](https://developer.mozilla.org/en/docs/Web/API/console) says `debug` is deprecated and it lists `exception` only as `_exception`. I think some more research into browser behaviour is needed.

Fishrock123 commented:
See https://github.com/nodejs/node/pull/3584#issuecomment-152557274, the situation is generally very silly.

jasnell commented:
Closing due to lack of activity. Seems like there's very little interest in this in general.
